pkg/asset/manifests/scheduler.go

```diff
@@ -44,7 +48,29 @@ func (s *Scheduler) Generate(dependencies asset.Parents) error {
 			Name: "cluster",
 			// not namespaced
 		},
-		Spec: configv1.SchedulerSpec{},
+		Spec: configv1.SchedulerSpec{
+			MastersSchedulable: false,
+		},
+	}
+
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+	computeReplicas := int64(0)
+	for _, pool := range installConfig.Config.Compute {
+		if pool.Replicas != nil {
+			computeReplicas += *pool.Replicas
+		}
+	}
+	if computeReplicas == 0 {
+		// A schedulable host is required for a successful install to complete.
+		// If the install config has 0 replicas for compute hosts, it's one of two cases:
+		//   1. An IPI deployment with no compute hosts.  The deployment can not succeed
+		//      without MastersSchedulable = true.
+		//   2. A UPI deployment.  The deployment may add compute hosts, but to ensure the
+		//      the highest probability of a successful deployment, we default to
+		//      schedulable masters.
```

Comment:
The trade-off here I guess is between:

- How painful is it to configure for (2) as a post-install (i.e. we made an incorrect assumption, and the user needs to correct for that) versus
- How much do we want to avoid an install-config option that allows the user to express their intent here - i.e. openshift/installer#2024

I would imagine that in the case of (2) we might find users just leave masters as schedulable even after they add workers?

IMO it's important enough for us to understand the intent that we should add the install-config option